digdag-standards/src/main/java/io/digdag/standards/operator/jdbc/AbstractJdbcOperator.java

```diff
@@ -0,0 +1,224 @@
+            }
+            queryId = state.get(QUERY_ID, UUID.class);
+        }
+
+        try (JdbcConnection connection = connect(connectionConfig)) {
+            Exception statementError = connection.validateStatement(query);
+            if (statementError != null) {
+                throw new ConfigException("Given query is invalid", statementError);
+            }
+
+            if (readOnlyMode) {
+                if (downloadFile.isPresent()) {
+                    connection.executeReadOnlyQuery(query, (results) -> downloadResultsToFile(results, downloadFile.get()));
+                }
+                else {
+                    connection.executeReadOnlyQuery(query, (results) -> skipResults(results));
+                }
+                return TaskResult.defaultBuilder(request).build();
+            }
+            else {
+                String statement;
+                boolean statementMayReturnResults;
+                if (insertInto.isPresent() || createTable.isPresent()) {
+                    if (insertInto.isPresent()) {
+                        statement = connection.buildInsertStatement(query, insertInto.get());
+                    }
+                    else {
+                        statement = connection.buildCreateTableStatement(query, createTable.get());
+                    }
+
+                    Exception modifiedStatementError = connection.validateStatement(statement);
+                    if (modifiedStatementError != null) {
+                        throw new ConfigException("Given query is valid but failed to build INSERT INTO statement (this may happen if given query includes multiple statements or semicolon \";\"?)", modifiedStatementError);
+                    }
+                    statementMayReturnResults = false;
+                    logger.debug("Running a modified statement: {}", statement);
+                }
+                else {
+                    statement = query;
+                    statementMayReturnResults = true;
+                }
+
+                TransactionHelper txHelper;
+                if (strictTransaction) {
+                    txHelper = connection.getStrictTransactionHelper(statusTableName);
+                }
+                else {
+                    txHelper = new NoTransactionHelper();
+                }
+
+                txHelper.prepare();
+
+                boolean executed = txHelper.lockedTransaction(queryId, () -> {
+                    if (statementMayReturnResults) {
+                        connection.executeScript(statement);
+                    }
+                    else {
+                        connection.executeUpdate(statement);
+                    }
+                });
+
+                if (!executed) {
+                    logger.debug("Query is already completed according to status table. Skipping statement execution.");
+                }
+
+                try {
+                    txHelper.cleanup();
```

Comment:
> delayed more than 24 hours

I see. That may happen. It seems better to store status on digdag status params as well.